docs/content/advanced/backend/entities.md

```diff
@@ -0,0 +1,158 @@
+
+Aside from Medusa’s core entities, you can also create your own entities to use in your Medusa server. Custom entities must reside in the `src/models` directory of your Medusa server.
+
+Entities are TypeScript files and they are based on [Typeorm’s Entities](https://typeorm.io/entities) and use Typeorm decorators.
+
+All entities must extend either the `BaseEntity` or `SoftDeletableEntity` classes. The `BaseEntity` class holds common columns including the `id`, `created_at`, and `updated_at` columns. It also includes the method that takes care of automatically generating the ID field of your entity.
+
+The `SoftDeletableEntity` class extends the `BaseEntity` class and adds another column `deleted_at`. If an entity can be soft deleted, meaning that a row in it can appear to the user as deleted but still be available in the database, it should extend `SoftDeletableEntity`.
+
+## How to Create a Custom Entity
+
+### Prerequisites
+
+It’s recommended to create a `tsconfig.json` file in the root of your Medusa server with the following content:
+
+```jsx
+{
+  "compilerOptions": {
+    "experimentalDecorators": true
+  }
+}
+```
+
+This will remove any errors that show up in your IDE related to experimental decorators.
+
+### Create the Entity
+
+To create an entity, create a TypeScript file in `src/models`. For example, here’s a `Post` entity defined in the file `src/models/post.ts`:
+
+```tsx
+import { BeforeInsert, Column, Entity, PrimaryColumn } from "typeorm";
+import { BaseEntity} from "@medusajs/medusa/dist/interfaces/models/base-entity";
+import { generateEntityId } from "@medusajs/medusa/dist/utils/generate-entity-id"
+
+@Entity()
+export class Post extends BaseEntity {
+  @Column({type: 'varchar'})
+  title: string | null;
+
+  @BeforeInsert()
+  private beforeInsert(): void {
+    this.id = generateEntityId(this.id, "post")
+  }
+}
+```
+
+This entity has one column `title` defined. However, since it extends `BaseEntity` it will also have the `id`, `created_at`, and `updated_at` columns.
+
+Medusa’s core entities all have the following format for IDs: `<PREFIX>_<RANDOM>`. For example, an order might have the ID `order_01G35WVGY4D1JCA4TPGVXPGCQM`.
+
+To generate an ID for your entity that matches the IDs generated for Medusa’s core entities, you should add a `BeforeInsert` event handler. Then, inside that handler use Medusa’s utility function `generateEntityId` to generate the ID. It accepts the ID as a first parameter and the prefix as a second parameter. The `Post` entity IDs will be of the format `post_<RANDOM>`.
+
+If you want the entity to also be soft deletable then it should extend `SoftDeletableEntity` instead:
+
+```tsx
+import { SoftDeletableEntity } from "@medusajs/medusa/dist/interfaces/models/soft-deletable-entity";
+
+@Entity()
+export class Post extends SoftDeletableEntity {
```

Comment:
Missing the closing }

Reply:
I know this is just to showcase where the change should be made but I guess I should make it clearer